x-pack/plugins/rollup/server/lib/search_strategies/rollup_search_strategy.js

```diff
@@ -0,0 +1,36 @@
+/*
+* Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+* or more contributor license agreements. Licensed under the Elastic License;
+* you may not use this file except in compliance with the Elastic License.
+*/
+import { callWithRequestFactory } from '../call_with_request_factory';
+
+const ROLLUP_INDEX_CAPABILITIES_METHOD = 'rollup.rollupIndexCapabilities';
+const INDEX_PATTERN_SEPARATOR = ',';
+
+export default (AbstractSearchStrategy, RollupSearchRequest) =>
+  (class RollupSearchStrategy extends AbstractSearchStrategy {
+    name = 'rollup';
+    batchRequestsSupport = false;
+
+    constructor(server) {
+      super(server, callWithRequestFactory, RollupSearchRequest);
+    }
+
+    async isRollupJobExists(req, indexPattern) {
+      const callWithRequest = this.getCallWithRequestInstance(req);
+      const indices = (indexPattern || '').split(INDEX_PATTERN_SEPARATOR);
+
+      const requests = indices.map(index => callWithRequest(ROLLUP_INDEX_CAPABILITIES_METHOD, {
+        indexPattern: index,
+      }));
+
+      return Promise.all(requests)
+        .then((responses) => responses.some((response, index) => Boolean(response[indices[index]])))
+        .catch(() => Promise.resolve(false));
+    }
+
+    async isViable(req, indexPattern) {
+      return await await this.isRollupJobExists(req, indexPattern);
```

Comment:
if I’m not wrong isRollupJobExists returns a promise that resolves to true or false, so 1 await is enough